client/src/extension.ts

```diff
@@ -211,7 +211,23 @@ export async function activate (context: vscode.ExtensionContext): Promise<void>
 
   logger.info('Congratulations, your extension "BitBake" is now active!')
 
-  void vscode.commands.executeCommand('bitbake.rescan-project')
+  // Update the scan result stored in the global state
+  bitBakeProjectScanner.onChange.on('scanReady', (bitbakeScanResult: BitbakeScanResult) => {
+    void context.workspaceState.update('bitbake.ScanResult', bitbakeScanResult).then(() => {
+      logger.debug('BitBake scan result saved to global state')
+    })
+  })
+
+  const lastBitbakeScanResult: BitbakeScanResult | undefined = context.workspaceState.get('bitbake.ScanResult')
+
+  if (lastBitbakeScanResult === undefined) {
+    logger.debug('No previous scan result found, rescanning the project')
+    void vscode.commands.executeCommand('bitbake.rescan-project')
+  } else {
+    logger.debug('Loading previous scan result')
+    bitBakeProjectScanner.onChange.emit('scanReady', lastBitbakeScanResult)

```

Comment:
I think we should also write the scan result in bitBakeProjectScanner.scanResult. It's used in several places, notably the selectRecipe function from the commands.

Reply:
done. check the new commit